### DISCUSSION TITLE
Enable JIT without enabling extra syntax/arbitrary values

### DISCUSSION THREAD

**Discussion by jmckenney** [tailwindlabs/tailwindcss#4148] [Ideas]
We love the new JIT compiler, but we've got a complex multi-tenant situation where the underlying platform needs careful fine-tuning and control of the config which is then extended by multiple brands. If we had a way to turn the new JIT syntax off/on that would be sweet.

For example, we need to be able to turn off support for w-[762px] syntax.

We're currently building this system for Williams-Sonoma brands (WS, West Elm, Pottery Barn /kids/teen, Mark & Graham).

Thanks for considering!

**Comment by TheRoadSouth** [comment]
+1 For large enterprise level applications, the ability to control some of the new functionality and syntax that comes with JIT would be awesome.

**Comment by MartinHughes-BPC** [comment]
Surely this should be internal company policy? If it were possible to do this, what would stop your staff from using custom style tags instead?

I'm just thinking: have a plan and talk to your colleagues.

**Reply by quick007** [reply]
Actually asked them this on discord (there both colleges) 
![image](https://user-images.githubusercontent.com/69743171/115628142-07e28c80-a2b5-11eb-938c-779d6e3bff63.png)

**Reply by TheRoadSouth** [reply]
It's easy to lint for custom style tags... it's more difficult to lint for dynamically stacked classes. The regex pattern for linting would need to be broad and could possible cause unintended false positives. Also, because of our multi tenant environment, style tags are basically useless (no way to use them dynamically for six brands).

**Reply by TheRoadSouth** [reply]
Also, we have hundreds of colleagues in many time zones around the world. It's next to impossible to keep tabs on what they are doing, especially considering we use external teams as well. We need to rely on automation and linting, or configuration, for a case like this.

**Reply by Martleby** [reply]
This doesn't make any sense to me. You have hundreds of colleagues building one central CSS file? If your requirements are that strict, why not just build a CSS file centrally, distribute that and say  no alterations or else?

**Reply by TheRoadSouth** [reply]
No, not at all. We have hundreds of devs building a massive eCommerce platform, split into many applications with their own CSS files, and then each of those further split by brand into a multi tenant architecture. It's a complex application that underpins a wide range of functionality for numerous brands, and thus needs a high level of configuration and control.

**Reply by TheRoadSouth** [reply]
To give you a couple of examples, we can't use the w-[762px] syntax, because while that specific value might work for one brand, it's unlikely to work for others. When we set up our Design System values, we have platform value defaults that are then overridden (or not) by multiple brands. So the configurability of Tailwind has been great. For a number of other reasons, it would be difficult for us to use the JIT compiler in prod until it reaches a stable point. But we could begin to use it for dev, and benefit from the speed, if we had a way to turn off the syntax that is not supported by the non-JIT TW which we need to stick to in prod for the time being.
